docs/ReduxIntegration.md

```diff
@@ -0,0 +1,377 @@
+
+export default connect(mapStateToProps, mapDispatchToProps)(AddTodo);
+```
+
+For `./components/TodoList.js`
+
+```jsx
+// TodoList is responsible for reading from Redux. Each row has a button which when pressed dispatches the deleteTodo method
+import React from 'react';
+import { FlatList, Text } from 'react-native';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { removeTodo } from '../actions/todoActions';
+
+function TodoElem({ todo, onDelete }) {
+  return (
+    <View style={styles.elemContainer}>
+      <View style={styles.textContainer}>
+        <Text style={styles.text}>{todo.text}</Text>
+        <Text style={styles.date}>{new Date(todo.date).toDateString()}</Text>
+      </View>
+      <View style={styles.buttonContainer}>
+        <Button onPress={() => onDelete(todo.id)} title="Delete" />
+      </View>
+    </View>
+  );
+}
+
+export function TodoList(props) {
+  const onDeleteTodo = (id) => props.removeTodo(id);
+
+  return (
+    <FlatList
+      data={props.todos}
+      keyExtractor={(todo) => todo.id.toString()}
+      renderItem={({ item }) => (
+        <TodoElem todo={item} onDelete={onDeleteTodo} />
+      )}
+    />
+  );
+}
+
+const mapStateToProps = (state) => ({
+  todos: state.todos,
+});
+
+const mapDispatchToProps = (dispatch) =>
+  bindActionCreators({ removeTodo }, dispatch);
+
+export default connect(mapStateToProps, mapDispatchToProps)(TodoList);
+```
+
+For `./App.js`
+
+```jsx
+import React from 'react';
+import { StyleSheet, View, StatusBar } from 'react-native';
+import { Provider } from 'react-redux';
+import configureStore from './store';
+import AddTodo from './components/AddTodo';
+import TodoList from './components/TodoList';
+
+const store = configureStore();
+
+export default function App() {
+  return (
+    <Provider store={store}>
+      <View style={styles.container}>
+        <StatusBar barStyle="dark-content" />
+        <AddTodo />
+        <TodoList />
+      </View>
+    </Provider>
+  );
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    paddingTop: 32,
+  },
+});
+```
+
+## Setting up the tests
+
+Now for the test part. The whole idea behind this is to test out the components that either dispatch actions on the redux store or read some data from the redux store. This means we will test `./components/TodoElem.js` and `./components/TodoList.js`. Thus we will create `./components/AddTodo.test.js` and `./components/TodoList.test.js`
+
+For `./components/AddTodo.test.js`
+
+```jsx
+import React from 'react';
+import { Provider } from 'react-redux';
+import { cleanup, fireEvent, render } from 'react-native-testing-library';
+import configureStore from '../store';
+import AddTodo from './AddTodo';
+
+describe('AddTodo component test', () => {
+  test('adds a new TODO when the button is pressed', () => {
+    const store = configureStore();
+
+    const component = (
+      <Provider store={store}>
+        <AddTodo />
```

Comment:
I think it would be nice to provide comments on why we wrap the component in a Provider for tests, where do we get `store` from, and that it's original but can also be mocked – the crucial thing is that it should use the real reducers we have, to resemble real environment as much as possible